app/controllers/candidate_api/candidates_controller.rb

```diff
@@ -0,0 +1,52 @@
+module CandidateAPI
+  class CandidatesController < ActionController::API
+    include ServiceAPIUserAuthentication
+
+    rescue_from ActionController::ParameterMissing, with: :parameter_missing
+    rescue_from ParameterInvalid, with: :parameter_invalid
+
+    # Makes PG::QueryCanceled statement timeout errors appear in Skylight
+    # against the controller action that triggered them
+    # instead of bundling them with every other ErrorsController#internal_server_error
+    rescue_from ActiveRecord::QueryCanceled, with: :statement_timeout
+
+    def index
+      render json: { data: serialized_candidates }
+    end
+
+    def parameter_missing(e)
+      error_message = e.message.split("\n").first
+      render json: { errors: [{ error: 'ParameterMissing', message: error_message }] }, status: :unprocessable_entity
+    end
+
+    def parameter_invalid(e)
+      render json: { errors: [{ error: 'ParameterInvalid', message: e }] }, status: :unprocessable_entity
+    end
+
+    def statement_timeout(e)
+      render json: { errors: [{ error: 'QueryCanceled', message: e }] }, status: :internal_server_error
```

Comment:
I think if we render a 500 we shouldn't expose the contents of the exception to the public — this should be a generic 500 like "there is a problem with the service". I don't think the Vendor API does the right thing here. I'll raise a ticket about that.